docs/sandbox-permissions.rst

```diff
@@ -223,9 +223,82 @@ requested with ``--filesystem`` and are not available with
 
 Device access
 `````````````
+You can provide the following device permissions:
+
+========= ======================================================
+``dri``   Direct Rendering Interface. Necessary for GL.
+``kvm``   Kernel based Virtual Machine ``/dev/kvm``
+``shm``   Shared Memory in ``/dev/shm``.
+``input`` Input devices as exposed in ``/dev/input``. This includes game controllers. Since flatpak 1.15.6.
+``all``   All devices, including all of the above except ``shm``
+========= ======================================================
 
 While not ideal, ``--device=all`` can be used to access devices like
-controllers or webcams.
+generic USB or webcams.
+
+The ``input`` device permission was introduced in 1.15.6. Adding this
+permission will have no effect on prior flatpak versions, and therefor
+prevent using the application as intended. Expecting user to update
+flatpak to the latest version is not possible as some LTS distribution
+will never update. We can expect took take 5 years for most users to
+be on the correct version and by that time, there will be more
+additions as more granular device permission get added.
+
+Starting flatpak 1.16, there is a device fallback mechanism. The
+device fallback mechanism is meant to specify that a certain device
+permission is to be used as a fallback, and therefor, if flatpak knows
+about it, it can tighten device access as intended.
+
+Packagers should take great care in considering device permissions and
+use the fallback mechanism when available.
+
+In the case where input device access is need you could specify:
+
+.. code-block::
+
+   --device=input
+
+On the command line, with flatpak < 1.15.6, you will get an error. In
+the metadata, it will be ignored and the permission will not be
+granted. This will make the package non-functional and offer a poor
+user experience.
```

Comment:
This seems to be mostly the same as the previous part. I'd add one line of warning for not using the feature then another line for the new fallback

Reply:
reworked the whole section.